src/stores/messages/MessagesDataProviderStore.ts

```diff
@@ -295,10 +296,20 @@ export default class MessagesDataProviderStore implements MessagesDataStore {
 		}
 
 		if (messages.length !== 0) {
+			const previousMessages = messages.filter(
```

Comment:
i think we need something like this
```
const prevMessages = getPrevMessages(); // will get messages with lesser timestamp or target
const firstNextMessage = messages[messages.length - 1];

if (firstNextMessage && firstNextMessage.messageId === this.messages[0]?.messageId) {
	prevMessages.pop();
}

const nextMessages = prevMessages.length ? messages.slice(0, prevMessages.length) : messages;

let newMessagesList = prevMessages.length
	? [
			...nextMessages,
			...[this.messages[0], ...prevMessages].filter(isEventMessage),
			...this.messages.slice(1),
			]
	: [...nextMessages, ...this.messages];

```

Reply:
...[this.messages[0], ...prevMessages].filter(isEventMessage) this part I will change to ...prevMessages.filter(isEventMessage) because otherwise there is a copy of initial message[0] appears in the list

Reply:
and getPrevMessages I decided to not create as new private function and leave here, because there is a function with similar name getPreviousMessages